packages/wrangler/src/vectorize/client.ts

```diff
@@ -0,0 +1,172 @@
+import { fetchListResult, fetchResult } from "../cfetch";
+import { requireAuth } from "../user";
+import type { Config } from "../config";
+import type {
+	VectorizeIndex,
+	VectorizeIndexDetails,
+	VectorizeQueryOptions,
+	VectorizeVector,
+	VectorizeVectorMutation,
+} from "@cloudflare/workers-types";
+
+const jsonContentType = "application/json; charset=utf-8;";
+
+interface VectorizeIndexResult extends VectorizeIndexDetails {
+	readonly created_on: string;
+	readonly modified_on: string;
+}
+
+export async function createIndex(
+	config: Config,
+	body: object
+): Promise<VectorizeIndexResult> {
+	const accountId = await requireAuth(config);
+
+	return await fetchResult<VectorizeIndexResult>(
+		`/accounts/${accountId}/vectorize/indexes`,
+		{
+			method: "POST",
+			headers: {
+				"content-type": jsonContentType,
+			},
+			body: JSON.stringify(body),
+		}
+	);
+}
+
+export async function deleteIndex(
+	config: Config,
+	indexName: string
+): Promise<void> {
+	const accountId = await requireAuth(config);
+	return await fetchResult<void>(
```

Comment:
Note for reviewer: Our API returns a HTTP 204, which `fetchResult` does not like. This is semantically valid for a `DELETE`, but isn't a great experience for users:

```
➜  node bin/wrangler.js vectorize delete with-table-output-nine
```
```
Deleting Vectorize index with-table-output-nine
✔ ⚠️  OK to delete the index 'with-table-output-nine'? … yes

✘ [ERROR] Received a malformed response from the API

  
  DELETE /accounts/d458dbe698b8eef41837f941d73bc5b3/vectorize/indexes/with-table-output-nine -> 204
  No Content
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/workers-sdk/issues/new/choose
```

Reply:
Yeah this could probably be a fast follow, to improve the UX

Reply:
Addressed this specifically by catching the case where `jsonText === ""` and the HTTP status code is 204 or 205.